src/include/rbd/librbd.h

```diff
@@ -144,6 +144,8 @@ enum {
   RBD_IMAGE_OPTION_JOURNAL_ORDER = 5,
   RBD_IMAGE_OPTION_JOURNAL_SPLAY_WIDTH = 6,
   RBD_IMAGE_OPTION_JOURNAL_POOL = 7,
+  RBD_IMAGE_OPTION_FEATURES_SET = 8,
```

Comment:
Okey, Let me try to explain what I want to do here. Yes, if ```RBD_IMAGE_OPTION_FEATURES``` is set, we shoud use them, otherwise use the defaults. And where are the defaults from, two ways, one from g_conf, the other one from parent. But now, the logic is placed in librbd, rather than rbd tools, then when we want to specify features partially while using default, that's not easy, because we don't know what is the default features at that time.

Example: --image-shared. When we use clone with --image-shared. we want to use the features from parent, at the same time, we want to specify some of the features.

To solve this kind of problem, I introduced two options ```RBD_IMAGE_OPTION_FEATURES_SET``` and ```RBD_IMAGE_OPTION_FEATURES_CLEAR```.  They are working as a additional way for features. Then RBD_IMAGE_OPTION_FEATURES still work as you described, " If RBD_IMAGE_OPTION_FEATURES is set, you should use them -- otherwise use the defaults.". But if we want to do some partial specification, such as clone with --image-shared. we can use these two options. 

I hope I explain myself clearly.